apps/dashr-shiny-3devents/app.R

```diff
@@ -0,0 +1,81 @@
+appName <- Sys.getenv("DASH_APP_NAME")
+if (appName != "") {
+  pathPrefix <- sprintf("/%s/", appName)
+  Sys.setenv(
+    DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+    DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix
+  )
+  setwd(sprintf("/app/apps/%s", appName))
+}
+
+library(dash)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(plotly)
+
+app <- Dash$new()
+
+app$layout(
+  htmlA(list(
+    htmlImg(id = "banner-image", src = "assets/image.png")
+  ), className = "logo",
+  href = "https://dashr.plot.ly/"),
+  htmlDiv(list(
+    dccGraph(
+      id = '3dplot',
+      figure = plot_ly(
+        x = rnorm(10),
+        y = rnorm(10),
+        z = rnorm(10),
+        type = "scatter3d"
+      ),
+      clear_on_unhover = TRUE
+    ),
+    htmlDiv(id = "hoverdataOutput"),
+    htmlBr(),
+    htmlDiv(id = "clickdataOutput")
+  ))
+)
+
+app$callback(
+  output = list(id = "hoverdataOutput", property = "children"),
+  params = list(input(id = "3dplot", property = "hoverData")),
+  function(hover_data) {
+    if (is.null(hover_data[[1]])) {
+      d <- "Hover events appear here (unhover to clear)"
+    } else {
+      d <- paste("Curve Number:", hover_data[[1]][[1]]$curveNumber,
+            "Point Number:", hover_data[[1]][[1]]$pointNumber,
+            "x:", hover_data[[1]][[1]]$x,
+            "y:", hover_data[[1]][[1]]$y,
+            "z:", hover_data[[1]][[1]]$z)
+    }
+    return(d)
+  }
+)
+
+
+app$callback(
+  output = list(id = "clickdataOutput", property = "children"),
+  params = list(input(id = "3dplot", property = "clickData")),
+  function(click_data) {
+    if (is.null(click_data[[1]])) {
+      d <- "Click events appear here (double-click to clear)"
```

Comment:
It doesn't seem like double-click to clear is working for the `clickData` in this case. Double-clicking on the point in question, or the graph, do not clear the output.